### PR TITLE
Add support for HTTP PATCH method

### PR DESCRIPTION
**Pull request by JGulbronson**
Need to fix up the formatting, and look at adding an enum for HTTP methods in DispatchMechanism.

### CODE REVIEW

**Comment by JGulbronson** [comment]
I mostly just followed what was done in #1369 

However, since Jetty doesn't have built-in support for PATCH the same way it does for other verbs, I had to hack that on a bit.

**Comment by JGulbronson** [comment]
@swankjesse I can't seem to add reviewers for some reason, but would love your thoughts on this.

File: misk/src/main/kotlin/misk/web/DispatchMechanism.kt [hunk 1]
```diff
@@ -19,19 +18,20 @@ enum class DispatchMechanism {
    * This can be misleading: web sockets don't behave like normal GETs (they are upgraded), and GRPC
    * calls don't behave like normal POSTS (they are duplex).
    */
-  val method: HttpMethod
+  val method: String
```
Comment (JGulbronson) [inline misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: We possibly want this as a new enum to avoid comparing strings.

**Comment by JGulbronson** [comment]
It seems I'm having trouble reproducing the failing tests locally

File: misk/src/main/kotlin/misk/web/DispatchMechanism.kt (hunk 1 above)
Comment (shellderp) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: do we really need that if `DispatchMechanism` is already an enum? also this value could be a constructor param on this enum instead

Comment (JGulbronson) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: That's a good point, method is really just a subset of `DispatchMechanism`, so method could probably return a `DispatchMechanism` instead for cases like `WEBSOCKET` and `GET`

Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: Why not use HttpMethod?

File: misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt [hunk 2]
```diff
@@ -59,6 +59,13 @@ internal class WebActionsServlet @Inject constructor(
     }
   }
 
+  override fun service(request: HttpServletRequest?, response: HttpServletResponse?) {
+    if (request?.method == "PATCH" && response != null) {
+      doPatch(request, response)
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt:64]: You need a `return` here

File: misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt [hunk 3]
```diff
@@ -59,6 +59,13 @@ internal class WebActionsServlet @Inject constructor(
...
+    }
+    super.service(request, response)
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt:66]: should we rename `handleCall` to `service` and remove the doGet/doPost/doDelete boilerplate?

File: misk/src/main/kotlin/misk/web/DispatchMechanism.kt (hunk 1 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: OMG, it’s because HttpMethod doesn’t have PATCH! Craxy

Comment (JGulbronson) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: Yeah, Jetty stuck a little too close to the spec on that one. Should we keep it as a string? Or try to use an enum?

File: misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt (hunk 3 above)
Comment (JGulbronson) [reply misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt:66]: Yeah maybe, though follow-up PR?

File: misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt (hunk 2 above)
Comment (JGulbronson) [reply misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt:64]: I'm actually confused how the test passes, it should be calling `doPatch` and then end up throwing later in the function. Will debug locally to make sure something else isn't happening.

Comment (JGulbronson) [reply misk/src/main/kotlin/misk/web/jetty/WebActionsServlet.kt:64]: Yeah Jetty throws this in the background but the tests happily passes
```
java.lang.IllegalStateException: COMPLETED
	at org.eclipse.jetty.server.HttpChannelState.sendError(HttpChannelState.java:893)
	at org.eclipse.jetty.server.Response.sendError(Response.java:461)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:732)
	at org.eclipse.jetty.websocket.servlet.WebSocketServlet.service(WebSocketServlet.java:176)
	at misk.web.jetty.WebActionsServlet.service(WebActionsServlet.kt:66)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:755)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:547)
```

Adding the `return` fixes

**Comment by JGulbronson** [comment]
@swankjesse thanks for the review! Added the `return`, it's a little concerning that exception deep down in Jetty didn't fail the test. Happy to add a new enum if we want that for `method()`, and then remove the boilerplate in a followup if that's the route we want to take.

File: misk/src/main/kotlin/misk/web/DispatchMechanism.kt (hunk 1 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/DispatchMechanism.kt:21]: String is fine.

**Pull request merged**